pkg/lca/ibu_test.go

```diff
@@ -0,0 +1,693 @@
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestImageBasedUpgradeWithSeedImageVersion(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		seedImageVersion    string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+			seedImageVersion:    "seedversion",
+		},
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+			seedImageVersion:    "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testIBU := generateImageBasedUpgrade()
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testIBU)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		ibuBuilder, err := PullImageBasedUpgrade(testSettings)
+		assert.Nil(t, err)
+
+		// Test the WithExtraManifests function
+		builderResult := ibuBuilder.WithSeedImageVersion(
+			testCase.seedImageVersion)
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestImageBasedUpgradeWithSeedImagePullSecretRef(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		pullSecretName      string
+	}{
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+			pullSecretName:      "pull-secret",
+		},
+		{
+			expectedError:       nil,
+			addToRuntimeObjects: true,
+			pullSecretName:      "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		var (
+			runtimeObjects []runtime.Object
+			testSettings   *clients.Settings
+		)
+
+		testIBU := generateImageBasedUpgrade()
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testIBU)
+		}
+
+		testSettings = clients.GetTestClients(clients.TestClientParams{
+			K8sMockObjects: runtimeObjects,
+		})
+
+		ibuBuilder, err := PullImageBasedUpgrade(testSettings)
+		assert.Nil(t, err)
+
+		// Test the WithExtraManifests function
+		builderResult := ibuBuilder.WithSeedImagePullSecretRef(
+			testCase.pullSecretName)
+
+		// Check the error
+		assert.Equal(t, err, testCase.expectedError)
+
+		if testCase.expectedError == nil {
+			assert.NotNil(t, builderResult)
+		}
+	}
+}
+
+func TestImageBasedUpgradeWaitUntilStageComplete(t *testing.T) {
+	testCases := []struct {
+		expectedError       error
+		addToRuntimeObjects bool
+		expectedErrorText   string
+		stage               string
+	}{
+		{
+			expectedError:       fmt.Errorf("wrong stage selected for imagebasedupgrade"),
+			addToRuntimeObjects: true,
+			stage:               "wrong",
+		},
+		{
+			expectedError:       fmt.Errorf("wrong stage selected for imagebasedupgrade"),
+			addToRuntimeObjects: true,
+			stage:               "",
+		},
+	}
```

Comment:
You should include a positive test no?

Reply:
A positive test will cause a jump into the PollUntilContextTimeout which takes 30 minutes

Reply:
It appears to me that it would take just a second if you use `buildTestClientWithDummyObject`

Reply:
Updated. Thanks.